File: 26-remove-duplicates-from-sorted-array/remove-duplicates-from-sorted-array.cpp
class Solution {
public:
    int removeDuplicates(vector<int>& arr) {
        // if(arr.empty()){
        //     return 0;
        // }
        // int index=1;
        // int n=arr.size();
        // for(int i=1;i<n;i++){
        //     if(arr[i]!=arr[i-1]){
        //       arr[index++]=arr[i];
        //     }
        // }
        // return index;
        set<int>st;
        int n=arr.size();
        for(int i=0;i<n;i++){
            st.insert(arr[i]);

        }
        int k=st.size();
        int j=0;
        for(auto i :st){
            arr[j]=i;
            j++;
        }
        return j;

    }
};

File: 26-remove-duplicates-from-sorted-array/remove-duplicates-from-sorted-array.java
class Solution {
    public int removeDuplicates(int[] nums) {
        HashSet<Integer>hs=new HashSet<>();
        int write=0;
        for(int i=0;i<nums.length;i++){
            if(!hs.contains(nums[i])){
                hs.add(nums[i]);
            
                nums[write++]=nums[i];
            }
        }
        return write;
        
    }
}